pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -366,6 +366,11 @@ def default_compilation_pass(self, optimisation_level: int = 2) -> BasePass:
         ]
         squash = auto_squash_pass({OpType.PhasedX, OpType.Rz})
 
+        def flatten_registers(c: "Circuit") -> "Circuit":
```

Comment:
It would be good to add a comment explaining why we are doing this. Particularly why it is necessary here but not in other backends (or is it?).

Reply:
I can do if you think the QASM explanation is lacking though!